e2e/publish_test.go

```diff
@@ -279,4 +281,164 @@ spec:
 		stdout, stderr, err := kubectl("delete", "logicalvolume", "csi-node-test-block")
 		Expect(err).ShouldNot(HaveOccurred(), "stdout=%s, stderr=%s", stdout, stderr)
 	})
+
+	It("should publish filesystem with mount option", func() {
+		By("creating a PVC and Pod")
+		lvYaml := []byte(`kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: topo-pvc-mount-option
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+  storageClassName: topolvm-provisioner-mount-option
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: ubuntu-mount-option
+  labels:
+    app.kubernetes.io/name: ubuntu
+spec:
+  containers:
+    - name: ubuntu
+      image: quay.io/cybozu/ubuntu:20.04
+      command: ["/usr/local/bin/pause"]
+      volumeMounts:
+        - mountPath: /test1
+          name: my-volume
+  volumes:
+    - name: my-volume
+      persistentVolumeClaim:
+        claimName: topo-pvc-mount-option
+`)
+
+		_, _, err := kubectlWithInput(lvYaml, "apply", "-f", "-")
+		Expect(err).ShouldNot(HaveOccurred())
+
+		Eventually(func() error {
+			stdout, stderr, err := kubectl("get", "pod", "ubuntu-mount-option", "-o", "yaml")
+			if err != nil {
+				return fmt.Errorf("failed to get pod. stdout: %s, stderr: %s, err: %v", stdout, stderr, err)
+			}
+
+			var pod corev1.Pod
+			err = yaml.Unmarshal(stdout, &pod)
+			if err != nil {
+				return err
+			}
+
+			if pod.Status.Phase != corev1.PodRunning {
+				return errors.New("Pod is not running")
+			}
+
+			return nil
+		}).Should(Succeed())
+
+		By("check mount option")
+		stdout, stderr, err := kubectl("get", "pvc", "topo-pvc-mount-option", "-o", "yaml")
+		Expect(err).ShouldNot(HaveOccurred(), "stdout=%s, stderr=%s", stdout, stderr)
+
+		var pvc corev1.PersistentVolumeClaim
+		err = yaml.Unmarshal(stdout, &pvc)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		fmt.Printf("pvc stdout: %+v\n", string(stdout))
+		fmt.Printf("pvc: %+v\n", pvc)
+		fmt.Printf("pvc.Spec.ClaimRef: %+v\n", pvc.Spec.VolumeName)
```

Comment:
Is this code for temporary debugging?
If for logging, use [ginkgo.writer](https://pkg.go.dev/github.com/onsi/ginkgo@v1.15.0/internal/writer) for better logging.

Reply:
Thank you. I fixed it.